pkg/parser/multiline.go

```diff
@@ -0,0 +1,84 @@
+package parser
+
+import "strings"
+
+// multilineItem is used to represent multiline array and map literals
+// as used in formatting. It does not explicitly store array or map
+// values, just placeholders - "el" for next element in arrays, keys for
+// maps.
+//
+// A multilineItem can be:
+// 1. Empty `""` representing a newline.
+// 2. Starting with `"//"` represents a full comment.
+// 3a. For arrays the multiline item `"el"` represents the next array
+// literal element.
+// 3b. For maps all the multiline items that don't represent
+// newline (`""`) or comments (`// …`) are the key of the next map pair.
+//
+// The follow example shows a multiline array literal and its
+// multilineItem slice representation:
+//
+//	arr := [
+//	    1 // commment1
+//	    2
+//
+//	     // comment2
+//	     3
+//	]
+//
+// has the representation:
+//
+//	[ "el", "// comment1", "el", "", "", "// comment2", "el", ""]
```

Comment:
Should the first slice element be `""` ? The first element is on the second line. With the map example below, it starts with `""` - yeah, looking at the test cases, this should start with `""`

Reply:
Thanks, fixed (I had it correct but then mucked it up in the re-format of the comment....)